src/main.cpp

```diff
@@ -88,21 +93,48 @@ int main(int argc, char** argv)
             test->run();
         } catch (tests::TestBase::TestAborted& e) {
             std::cout << test_name << " aborted early (" << e.what() << ")" << std::endl;
-            failed = true;
+            fails[i] = true;
         }
 
-        if (!failed) {
+        if (!fails[i]) {
             tests::TestBase::Result result = test->getResult();
             if (result != tests::TestBase::Result::Success) {
-                failed = true;
+                fails[i] = true;
             }
             std::cout << test_name << " test result: " << result << std::endl;
         }
 
         // store the actually used config (which includes the default values) back
         // in the YAML config node
         test->storeConfig(test_node);
+        std::cout << "----" << std::endl;
+        ++i;
+    }
+
+    std::cout << "Test summary:" << std::endl;
+
+    i = 0;
+    for (auto test_node : tests_node) {
+        std::cout << test_node["name"].as<std::string>() << ":" << std::endl;
+        for (auto config : test_node) {
+            std::cout << config.first.as<std::string>() << ": " << config.second.as<std::string>()
+                      << std::endl;
+        }
+
+        // TODO: this is ugly but otherwise clang-tidy complains.
```

Comment:
What's the complaint and how would you like it?

Reply:
I used the tertiary operator and I got this weird complaint:
```
error: do not implicitly decay an array into a pointer; consider using gsl::array_view or an explicit cast instead [cppcoreguidelines-pro-bounds-array-to-pointer-decay,-warnings-as-errors]
        std::cout << "  => " << (fails[i] ? "failed" : "passed") << std::endl;
```

Reply:
You can disable that warning here: https://github.com/Auterion/mavlink-testing-suite/blob/master/.clang-tidy#L1

Reply:
Worked around: 7a06ffe007ec8ab1a8c71c54823431505c167bc4.